src/univariate/discrete/negativebinomial2.jl

```diff
@@ -0,0 +1,121 @@
+"""
+Alternative location-scale parameterization of the negative binomial distribution
+in terms of location `μ` and overdispersion `ϕ`.
+
+```math
+P(X = n) = {n + \\phi - 1 \\choose n} (\\frac{\\mu}{\\mu + \\phi})^n (\\frac{\\phi}{\\mu + \\phi})^\\phi, \\quad \\text{for } n = 0,1,2,\\ldots
+```
+
+In terms of the mixture definition, this corresponds to:
+```math
+n \\sim \\text{Poisson}(z), \\quad
+z \\sim \\text{Gamma}(\\phi, \\frac{\\mu}{\\phi})
+```
+This provides a clear interpretation of the negative binomial as an overdispersed
+Poisson with expected location 𝔼[z] = μ and var[z] = μ + μ²/ϕ. It also has advantages in that
```

Comment:
Is there a specific reason why you're choosing this parametrisation, as opposed to `var[z] = μ + μ² * ϕ`?
this would make more sense in relation to calling ϕ the overdispersion parameter, as then larger ϕ -> larger dispersion, and ϕ = 0 would actually recover the `Poisson` distribution.

Reply:
Thank you for this suggestion. At first, I thought: why deviate from Stan? But then remembered [this](https://github.com/stan-dev/stan/wiki/Prior-Choice-Recommendations#story-when-the-generic-prior-fails-the-case-of-the-negative-binomial). Clearly, it's preferable to use `var[z] = μ + μ² * ϕ`, rather than `var[z] = μ + μ²/ϕ`, as one can then reason about overdispersion without invoking its inverse. Incidentally, Dan Simpson's [remarks](https://statmodeling.stat.columbia.edu/2018/04/03/justify-my-love/) support this line of reasoning.

But why might the Stan implementation prefer the inverse? Nothing that I can see from a numerical stability perspective. Other than a historical precedent, my guess might be that the pmf is uglier when parameterized such that `var[z] = μ + μ² * ϕ`.